docs/INSTALL.md

```diff
@@ -1,9 +1,69 @@
 # Building
 
+- [Getting sources](#getting-sources)
 - [Desktop](#desktop-app)
 - [Android](#android-app)
 - [iOS](#ios-app)
 
+## Getting sources
+
+First of all get the source code. The full Organic Maps sources repository is ~8.5Gb in size, however its possible to reduce the size e.g. if full commit history is not required.
+
+For _Windows 10_ enable [symlinks][git-symlinks] support in git first:
+
+[git-symlinks]: https://git-scm.com/docs/git-config#Documentation/git-config.txt-coresymlinks
+
+```bash
+git config --global core.symlinks true
+```
+
+If you plan to contribute and propose pull requests then fork the repo https://github.com/organicmaps/organicmaps first.
+
+Clone the main repository (~5.5Gb):
+
+```bash
+git clone https://github.com/organicmaps/organicmaps.git
+```
+or your fork of it:
+```bash
+git clone https://github.com/<your-github-name>/organicmaps.git
+```
+
+Add a `--filter=blob:limit=128k` option to above command to make a _partial clone_ of just ~1.2Gb, i.e. blob files over 128k in size will be excluded form the history and downloaded on-demand - should be suitable for a generic development use.
+
+And/or add a `--depth=1` option to make a _shallow copy_ (and possibly a `--no-single-branch` to have all branches not just `master`), i.e. omit history retaining current commits only - suitable for one-off builds.
```

Comment:
Let's put it into the clone cmd together with --recurse-submodules.

The install doc should show the simplest and the fastest way to build and run. All dev details can be stored in contributing doc. What do you think?

Reply:
I understand your idea of separating contribution and installation docs.

For the reference, at the moment `CONTRIBUTING.md` says:
> Code Contributions
> 
> Please follow instructions in INSTALL.md to set up your development environment. Create and submit a Pull Request with your changes.

I think its possible to separate but it'll mean a lot of duplication, basically the bulk of current `INSTALL.md` would be moved to `CONTRIBUTING.md` and then `INSTALL.md` simplified.

How do we define a build/run-only env? For that case I think there is no need for Android Studio and actually the fastest way to get the repo would be shallow/partial clone as it reduces download times drastically.

I'd rather prefer the current approach with a detailed doc that gives a default/straightforward way of doing things alongside some options so the one following it can make decisions on the go.

A compromise would be to hide most of the optional things in e.g. footnotes or a separate subsection so that the "default way" is more clear.

Please let me know your further thoughts on this!

Reply:
Current docs are hard for newcomers. It would be great if there will be an "easy and fast" way to compile and run the app on Android, iOS, and Desktops. Also, ideally, if someone wants to make a very simple code change/fix, it should also be very easy and clear. And in parallel, there could be a very detailed doc for those who want to get all details.

I don't know how to better organize it. Any ideas are welcome.

Reply:
Maybe make starting INSTALL sections simple, but move all details to sections below? And insert some links to them to avoid duplication?

Reply:
Yeap I agree it should be simple and newbie-friendly.
I'll try to re-arrange!